Doppler.Sap/Factory/CreditNoteHandler.cs

```diff
@@ -76,12 +85,99 @@ public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
             }
         }
 
+        public async Task<SapTaskResult> UpdatePaymentStatusHandle(SapTask dequeuedTask)
+        {
+            try
+            {
+                var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.CreditNoteRequest.BillingSystemId);
+                var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
+                var existentCreditNote = await sapTaskHandler.TryGetCreditNoteByCreditNoteId(dequeuedTask.CreditNoteRequest.CreditNoteId);
+
+                if (existentCreditNote != null)
+                {
+                    var sapResponse = await UpdateCreditNote(existentCreditNote, dequeuedTask, sapSystem);
+
+                    if (sapResponse.IsSuccessful)
+                    {
+                        if (dequeuedTask.CreditNoteRequest.Type == (int)CreditNoteEnum.Refund &&
+                            dequeuedTask.CreditNoteRequest.TransactionApproved)
+                        {
+                            return await SendOutgoingPaymentToSap(sapSystem, existentCreditNote, dequeuedTask.CreditNoteRequest.TransferReference);
+                        }
+                    }
+                    else
+                    {
+                        _logger.LogError($"Credit Note could'n update to SAP because exists an error: '{sapResponse.SapResponseContent}'.");
+                    }
+
+                    return sapResponse;
+                }
+                else
+                {
+                    return new SapTaskResult
+                    {
+                        IsSuccessful = false,
+                        SapResponseContent = $"Credit Note could'n update to SAP because the credit note does not exist: '{dequeuedTask.CreditNoteRequest.CreditNoteId}'.",
+                        TaskName = "Updating Credit Note Request"
+                    };
+                }
```

Comment:
Hemos visto que en general es conveniente inverter los if y reducir el nesting, incluso si eso implica repetir el checkeo de una condición, eso hace más explícito los flujos posibles:

Ejemplo:

```suggestion
                if (existentCreditNote == null)
                {
                    return new SapTaskResult
                    {
                        IsSuccessful = false,
                        SapResponseContent = $"Credit Note could'n update to SAP because the credit note does not exist: '{dequeuedTask.CreditNoteRequest.CreditNoteId}'.",
                        TaskName = "Updating Credit Note Request"
                    };
                }
                
                var sapResponse = await UpdateCreditNote(existentCreditNote, dequeuedTask, sapSystem);

                if (!sapResponse.IsSuccessful)
                {
                     _logger.LogError($"Credit Note could'n update to SAP because exists an error: '{sapResponse.SapResponseContent}'.");
                     return sapResponse;
                }
                else if (dequeuedTask.CreditNoteRequest.Type != (int)CreditNoteEnum.Refund || !dequeuedTask.CreditNoteRequest.TransactionApproved)
                {
                     _logger.LogError($"Credit Note could'n update to SAP because exists an error: '{sapResponse.SapResponseContent}'.");
                     return sapResponse;
                }
                else 
                {
                    return await SendOutgoingPaymentToSap(sapSystem, existentCreditNote, dequeuedTask.CreditNoteRequest.TransferReference);
                }
```

Reply:
excellent, good aproach, I will change it. Thanks @andresmoschini.

Reply:
Done